src/node/genesis_gen.h

```diff
@@ -398,10 +398,41 @@ namespace ccf
       shares_view->put(0, key_share_info);
     }
 
-    void set_recovery_threshold(size_t threshold)
+    bool set_recovery_threshold(size_t threshold)
     {
       auto config_view = tx.get_view(tables.config);
+
+      // While waiting for recovery shares, the recovery threshold cannot be
+      // modified. Otherwise, the threshold could be passed without triggering
+      // the end of recovery procedure
+      if (
+        get_service_status().value() ==
+        ServiceStatus::WAITING_FOR_RECOVERY_SHARES)
+      {
+        LOG_FAIL_FMT(
+          "Failed to set recovery threshold: service is currently waiting for "
+          "recovery shares");
+        return false;
+      }
+
+      if (threshold == 0)
+      {
+        LOG_FAIL_FMT("Recovery threshold cannot be set to 0");
```

Comment:
Why?

Reply:
Re-assembling a key from 0 shares is an open research problem ;)

I think the consequence of the threshold being set to zero would be that the ledger secret is effectively stored in clear in the ledger. Or stored encrypted with a key that's never been shared with anyone and the network is effectively irrecoverable.

The latter seems like a possible, if uncommon use case?

Reply:
This is a design "constraint" of the underlying secret sharing library: it is not possible to share a secret in `N` shares with a recovery threshold of `0`.

Reply:
I think a recovery threshold of 0 is trivially recoverable, but this is a plausible use case? The network offers no confidentiality, and it can be forked by any member (anyone who can start a new node and send a "try to recover" proposal, which will succeed without any shares). This is similar to a standard blockchain (you can fork whenever you like), but you get stronger binding of the _identity of the network_ which produced your favourite chain.

Of course you can almost do this with a threshold == 1, but then you're still restricted to 255 shares and members need to remember their share-decrypting keys; with 0 you could make everyone a member?

This is a blue-sky digression, feel free to ignore :)

Reply:
After some more discussion I'm convinced this trivially-forkable network is useless. The opposite interpretation (an unrecoverable network) may be applicable in some scenarios, but would require breaking more things (eg - code ID changes and governance changes) than just the recovery threshold.